tests/all/wasi_testsuite.rs

```diff
@@ -5,31 +5,97 @@
 
 #![cfg(not(miri))]
 
-use crate::cli_tests::run_wasmtime_for_output;
-use anyhow::Result;
+use crate::cli_tests::get_wasmtime_command;
+use anyhow::{anyhow, Result};
 use serde::Deserialize;
+use std::collections::HashMap;
 use std::ffi::OsStr;
-use std::fs::{self, DirEntry};
+use std::fs;
 use std::path::{Path, PathBuf};
-use std::process::Output;
+use std::process::{Command, Output};
+use walkdir::{DirEntry, WalkDir};
 
 #[test]
 #[cfg_attr(target_os = "windows", ignore)] // TODO: https://github.com/WebAssembly/WASI/issues/524
-fn wasi_threads_testsuite() -> Result<()> {
-    for module in list_modules("tests/wasi_testsuite/wasi-threads/test/testsuite")? {
-        println!("Testing {}", module.display());
-        let result = run(&module)?;
-        let spec = parse_spec(&module.with_extension("json"))?;
-        assert_eq!(spec, result);
+fn wasi_testsuite() -> Result<()> {
+    // Currently, Wasmtime's implementation in wasi-common does not line up
+    // exactly with the expectations in wasi-testsuite. This could be for one of
+    // various reasons:
+    //  - wasi-common has a bug
+    //  - wasi-testsuite overspecifies (or incorrectly specifies) a test
+    //  - this test runner incorrectly configures Wasmtime's CLI execution.
+    //
+    // The easiest way to resolve one of these is to remove the file from the
+    // list and run `cargo test wasi_testsuite -- --nocapture`. The printed
+    // output will show the expected result, the actual result, and a command to
+    // replicate the failure from the command line.
+    const WASI_COMMON_IGNORE_LIST: &[&str] = &[
```

Comment:
Should we add a comment about whether or not this list is expected to grow?

Reply:
I would expect this list to shrink &mdash; hopefully! Yeah, I can add something like that.